source/backend/api/Areas/Disposition/Controllers/DispositionFileController.cs

```diff
@@ -323,6 +323,80 @@ public IActionResult GetDispositionFileSales([FromRoute]long id)
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(DispositionFileAppraisalModel), 200)]
+        [SwaggerOperation(Tags = new[] { "dispositionfile" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public IActionResult GetDispositionFileAppraisal([FromRoute] long id)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(DispositionFileController),
+                nameof(GetDispositionFileAppraisal),
+                User.GetUsername(),
+                DateTime.Now);
+
+            _logger.LogInformation("Dispatching to service: {Service}", _dispositionService.GetType());
+
+            var dispositionSale = _dispositionService.GetDispositionFileAppraisal(id);
+            return new JsonResult(_mapper.Map<DispositionFileAppraisalModel>(dispositionSale));
+        }
+
+        [HttpPost("{id:long}/appraisal")]
+        [HasPermission(Permissions.DispositionEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(DispositionFileAppraisalModel), 201)]
+        [SwaggerOperation(Tags = new[] { "dispositionfile" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public IActionResult AddDispositionFileAppraisal([FromRoute] long id, [FromBody] DispositionFileAppraisalModel dispositionFileAppraisal)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(DispositionFileController),
+                nameof(AddDispositionFileAppraisal),
+                User.GetUsername(),
+                DateTime.Now);
+
+            _logger.LogInformation("Dispatching to service: {Service}", _dispositionService.GetType());
+
+            try
+            {
+                var dispositionAppraisalEntity = _mapper.Map<Dal.Entities.PimsDispositionAppraisal>(dispositionFileAppraisal);
+                var newDispositionAppraisal = _dispositionService.AddDispositionFileAppraisal(id, dispositionAppraisalEntity);
+
+                return new JsonResult(_mapper.Map<DispositionFileAppraisalModel>(newDispositionAppraisal));
+            }
+            catch (DuplicateEntityException e)
+            {
+                return Conflict(e.Message);
+            }
+        }
+
+        [HttpPut("{id:long}/appraisal/{appraisalId:long}")]
+        [HasPermission(Permissions.DispositionEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(DispositionFileAppraisalModel), 200)]
+        [SwaggerOperation(Tags = new[] { "dispositionfile" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public IActionResult UpdateDispositionFileAppraisal([FromRoute] long id, [FromRoute] long appraisalId, [FromBody] DispositionFileAppraisalModel dispositionFileAppraisal)
```

Comment:
To co-exist with our existing patterns I'd be tempted to remove the add endpoint entirely. In general when editing sub-entities we consider that an update. We only use post endpoints for first class entities like files or persons. sub-entities are all handled with puts, and that single put handles add/edit/delete (allthough in this case it'll just handle add/edit).